staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -1106,6 +1111,12 @@ func installAPI(s *GenericAPIServer, c *Config) {
 		}
 	}
 
+	if c.FeatureGate.Enabled(zpagesfeatures.ComponentFlagz) {
+		if c.Flagz != nil {
+			flagz.Install(s.Handler.NonGoRestfulMux, componentName, c.Flagz)
```

Comment:
is the component something to wire through from top-level? Now every apiserver will have the same constant name there.

Reply:
Yeah, realized that the name is already available to be used [here](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/server/config.go#L754), so wrirng that in the flagz handler.

Reply:
This change is making the response show "kube-aggregator" in the payload which probably makes sense since [this](https://github.com/kubernetes/kubernetes/blob/release-1.31/cmd/kube-apiserver/app/server.go#L168-L189) is how the server chain is created which puts the aggregator at the very last but this is the first server a request is made against. And /flagz request is not delegated down the chain.

Does it still make sense to have the name wired in like this or should I just revert back to the "kube-apiserver" const ?

Reply:
one fix for this could be to move the installation of the handler to [pkg/controlplane/apiserver/server.go](https://github.com/kubernetes/kubernetes/blob/release-1.31/pkg/controlplane/apiserver/server.go#L91), wdyt @sttts ?